core/common/src/main/java/alluxio/resource/DynamicResourcePool.java

```diff
@@ -341,8 +342,7 @@ public T acquire(long time, TimeUnit unit) throws TimeoutException, IOException
             throw new TimeoutException("Acquire resource times out.");
           }
         } catch (InterruptedException e) {
-          // Restore the interrupt flag so that it can be handled later.
-          Thread.currentThread().interrupt();
+          throw new IOException("Thread interrupted while acquiring client from pool: " + this);
```

Comment:
Whenever you react to thread interrupt and throw something else than `InterruptedException`, you should restore interrupt flag:
```
Thread.currentThread().interrupt();
```

Otherwise, the thread interrupt status gets lost and caller will likely not stop what's it's doing. 
For example, you can retry on some generic IOException, but you should not retry on InterruptedException. Likewise, you should not retry when thread is interrupted state.

Reply:
Should still restore the flag as @findepi suggested like this?

```
          Thread.currentThread().interrupt();
          throw new IOException("Thread interrupted while acquiring client from pool: " + this);
```

Reply:
Yeah I think this is the best practice when attempting to handle an interrupt. 
As a permanent fix, I prefer propagating the exception.